src/approval/approval.service.ts

```diff
@@ -0,0 +1,96 @@
+import Axios, { AxiosResponse, AxiosInstance } from 'axios';
+import { IApproverInfo, ICanApproveToUser, IApprovalRequest } from './approvers.interface';
+import { ApprovalError, NotFoundError, ApplicationError, UnauthorizedError } from '../utils/errors/errors';
+import { config } from '../config';
+import { getToken } from "../spike/spike.service";
+import { TransferRepository } from '../transfer/transfer.repository';
+
+export class ApprovalService {
+
+    private instance: AxiosInstance;
+
+    constructor() {
+        this.instance = Axios.create({ baseURL: config.approval.approvalUrl });
+        this.addAuthIntreceptor();
+    }
+
+    async createRequest(data: IApprovalRequest) {
+        try {
+            const res: AxiosResponse = await this.instance.post(`/api/v1/request`, data);
+            return res.data;
+        } catch (err) {
+            await TransferRepository.deleteByID(data.id || "");
+            if (!err.response || !err.response.status) throw new ApplicationError();
+
+            const status: number = err.response.status;
+            if (status === 401) {
+                throw new UnauthorizedError(`request was not authorized: ${err.message}`)
+            }
+            else if (status === 404) {
+                throw new NotFoundError(`The user is not found`);
+            } else if (status === 502) {
+                throw new ApprovalError(`Error was thrown by the approval service : ${err.message}`);
+            }
+            throw new ApprovalError(`Error in contacting the approval service : ${err.message}`);
+        }
+
+    }
+
+    /**
+     * Gets a user approver information from the approval service
+     * @param id - the user ID
+     */
+    async getApproverInfo(id: string): Promise<IApproverInfo> {
+        try {
+            const res: AxiosResponse = await this.instance.get(`/api/v1/users/${id}/approverInfo`);
+            const info: IApproverInfo = res.data;
+
+            return info;
+
+        } catch (err) {
+            if (err.response && err.response.status) {
+                const status: number = err.response.status;
+                if (status === 404) {
+                    throw new NotFoundError(`The user with id ${id} is not found`);
+                } else if (status === 502) {
+                    throw new ApprovalError(`Error was thrown by the approval service : ${err.message}`);
+                }
+                throw new ApprovalError(`Error in contacting the approval service : ${err.message}`);
+            } else {
+                throw new ApplicationError(`Unknown Error while contacting the approval service : ${err.message}`);
+            }
+        }
+    }
+
+    /**
+     * canApproveToUser checks if the user assign a good approver
```

Comment:
```suggestion
     * canApproveToUser checks if the user assigned a good approver
```

Reply:
It doesn't just check, it returns an object with information. You should detail it in the documentation.

Reply:
done.